src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingAssessmentResource.java

```diff
@@ -71,6 +89,112 @@ public ProgrammingAssessmentResource(AuthorizationCheckService authCheckService,
+        newResult.setParticipation(participation);
+
+        ProgrammingExercise exercise = (ProgrammingExercise) participation.getExercise();
+        checkAuthorization(exercise, user);
+
+        final var isAtLeastInstructor = authCheckService.isAtLeastInstructorForExercise(exercise, user);
+        if (!assessmentService.isAllowedToCreateOrOverrideResult(latestExistingResult.orElse(null), exercise, participation, user, isAtLeastInstructor)) {
+            log.debug("The user " + user.getLogin() + " is not allowed to override the assessment for the participation " + participation.getId() + " for User " + user.getLogin());
+            return forbidden("assessment", "assessmentSaveNotAllowed", "The user is not allowed to override the assessment");
+        }
+
+        final var course = exercise.getCourseViaExerciseGroupOrCourseMember();
+        if (!authCheckService.isAtLeastTeachingAssistantInCourse(course, user) || !exercise.areManualResultsAllowed()) {
+            return forbidden();
+        }
+
+        if (newResult.getResultString() == null) {
+            throw new BadRequestAlertException("Result string is required.", ENTITY_NAME, "resultStringNull");
+        }
+        else if (newResult.getResultString().length() > 255) {
+            throw new BadRequestAlertException("Result string is too long.", ENTITY_NAME, "resultStringNull");
+        }
+        else if (newResult.getScore() == null) {
+            throw new BadRequestAlertException("Score is required.", ENTITY_NAME, "scoreNull");
+        }
+        else if (newResult.getScore() < 100 && newResult.isSuccessful()) {
+            throw new BadRequestAlertException("Only result with score 100% can be successful.", ENTITY_NAME, "scoreAndSuccessfulNotMatching");
+        }
+        else if (!newResult.getFeedbacks().isEmpty() && newResult.getFeedbacks().stream().anyMatch(feedback -> feedback.getText() == null)) {
+            throw new BadRequestAlertException("In case feedback is present, feedback text and detail text are mandatory.", ENTITY_NAME, "feedbackTextOrDetailTextNull");
+        }
+
+        ProgrammingSubmission submission;
+        if (latestExistingResult.isEmpty()) {
+            // TODO: Double check if we can create a new result when the existing result has already a completion date (submitted)
+            // Create manual submission with last commit hash und current time stamp.
+            submission = programmingSubmissionService.createSubmissionWithLastCommitHashForParticipation((ProgrammingExerciseStudentParticipation) participation,
+                    SubmissionType.MANUAL);
+            newResult.setSubmission(submission);
+        }
+        else {
+            submission = programmingSubmissionService.findByIdWithEagerResultAndFeedback(latestExistingResult.get().getSubmission().getId());
+            newResult.setSubmission(submission);
+            if (newResult.getSubmission() == null) {
```

Comment:
Wouldn't it be easier to check the submission directly for null, before setting the submission of the newResult and getting it again? Just for readability

Reply:
Good catch! Fixed in